### DISCUSSION TITLE
Database Size / Reward & Commision History

### DISCUSSION THREAD

**Discussion by nico89-exp** [DeFiCh/app#935] [Q&A]
@izzycsy 

If my understanding is correct, the recent disk space saving of the blockchain data comes from pruning rewards and commissions history. If I want my full history back I need to reindex from scratch with checkpoints=0 in defi.conf. If my understanding is correct this not only creates the full wallet history but the full blockchain history. Of course I would only need my own full history.

Wouldn't it be possible to offer snapshots with full history in parallel and let the user choose in the app directly if he/she wants full or partial history? At least for own transactions, rewards and commissions.

Do I have a general issue in my understanding? Most likely ...

**Comment by gamsterx** [comment]
I just had the same idea and discovered your post too late. 😄 

Linking #940
